erigon-lib/state/inverted_index.go

```diff
@@ -983,6 +983,23 @@ func (ic *InvertedIndexContext) Prune(ctx context.Context, rwTx kv.RwTx, txFrom,
 		return fmt.Errorf("create %s keys cursor: %w", ii.filenameBase, err)
 	}
 	defer keysCursor.Close()
+
+	if !omitProgress {
+		pruneTxNum, _, err := GetExecV3PruneProgress(rwTx, ii.indexKeysTable)
+		if err != nil {
+			ic.ii.logger.Error("failed to get index prune progress", "err", err)
+		}
+		// pruning previously stopped at purunedTxNum; txFrom < pruneTxNum < txTo of previous range.
+		// to preserve pruning range consistency need to store or reconstruct pruned range for given key
+		// for InvertedIndices storing pruned key does not make sense because keys are just txnums,
+		// any key will seek to first available txnum in db
+		if pruneTxNum != 0 {
+			prevPruneTxFrom := (pruneTxNum / ii.aggregationStep) * ii.aggregationStep
+			prevPruneTxTo := prevPruneTxFrom + ii.aggregationStep
+			txFrom, txTo = prevPruneTxFrom, prevPruneTxTo
+		}
+	}
+
```

Comment:
I see next unhandled corner-case:
```
do keysTable.Delete() 
context cancel return
don't do any IdxTable.Delete()
can't resume IdxTable.Delete() in future - because keysTable already empty
```

Reply:
whoaaa i considered that case but you're right!
I thought that keys are collected into ETL and are NOT removed immediately but removed during `collector.Load()`. If we move a deletion in collector.Load() loop, will that close the problem?